src/map/itemdb.cpp

```diff
@@ -21,16 +21,999 @@
+}
+
+/**
+ * Reads and parses an entry from the item_db.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 ItemDatabase::parseBodyNode(const YAML::Node &node) {
+	uint32 nameid;
+
+	if (!this->asUInt32(node, "Id", nameid))
+		return 0;
+
+	std::shared_ptr<item_data> item = this->find(nameid);
+	bool exists = item != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "AegisName", "Name" }))
+			return 0;
+
+		item = std::make_shared<item_data>();
+		item->nameid = nameid;
+		item->flag.available = true;
+	}
+
+	if (this->nodeExists(node, "AegisName")) {
+		std::string name;
+
+		if (!this->asString(node, "AegisName", name))
+			return 0;
+
+		if (itemdb_search_aegisname(name.c_str())) {
+			this->invalidWarning(node["AegisName"], "Found duplicate item Aegis name for %s, skipping.\n", name.c_str());
+			return 0;
+		}
+
+		item->name.resize(ITEM_NAME_LENGTH);
+		item->name = name.c_str();
+	}
+
+	if (this->nodeExists(node, "Name")) {
+		std::string name;
+
+		if (!this->asString(node, "Name", name))
+			return 0;
+
+		item->ename.resize(ITEM_NAME_LENGTH);
+		item->ename = name.c_str();
+	}
+
+	if (this->nodeExists(node, "Type")) {
+		std::string type;
+
+		if (!this->asString(node, "Type", type))
+			return 0;
+
+		std::string type_constant = "IT_" + type;
+		int64 constant;
+
+		if (!script_get_constant(type_constant.c_str(), &constant) || constant < IT_HEALING || constant >= IT_MAX) {
+			this->invalidWarning(node["Type"], "Invalid item type %s for %s (%hu), defaulting to IT_ETC.\n", type.c_str(), item->name.c_str(), nameid);
+			constant = IT_ETC;
+		}
+
+		if (constant == IT_DELAYCONSUME) { // Items that are consumed only after target confirmation
+			item->type = IT_USABLE;
+			item->flag.delay_consume |= 0x1;
+		}
+
+		item->type = static_cast<item_types>(constant);
+	} else {
+		if (!exists)
+			item->type = IT_ETC;
+	}
+
+	if (this->nodeExists(node, "SubType")) {
+		std::string type;
+
+		if (!this->asString(node, "SubType", type))
+			return 0;
+
+		if (item->type == IT_WEAPON) {
+			std::string type_constant = "W_" + type;
+			int64 constant;
+
+			if (!script_get_constant(type_constant.c_str(), &constant) || constant < W_FIST || constant >= MAX_WEAPON_TYPE) {
+				this->invalidWarning(node["SubType"], "Invalid weapon type %s for %s (%hu), defaulting to W_FIST.\n", type.c_str(), item->name.c_str(), nameid);
+				item->subtype = W_FIST;
+			}
+		} else if (item->type == IT_AMMO) {
+			std::string type_constant = "AMMO_" + type;
+			int64 constant;
+
+			if (!script_get_constant(type_constant.c_str(), &constant) || constant <= AMMO_NONE || constant >= MAX_AMMO_TYPE) {
+				this->invalidWarning(node["SubType"], "Invalid ammo type %s for %s (%hu), defaulting to AMMO_NONE.\n", type.c_str(), item->name.c_str(), nameid);
+				item->subtype = AMMO_NONE;
+			}
+		} else
+			this->invalidWarning(node["SubType"], "SubType is not supported for this item type.\n");
```

Comment:
Did I miss something or `item->subtype` has no value when the constant is valid ?